### PR TITLE
Bug 1879648 - Disable menu toolbar when nav bar is enabled

### PR DESCRIPTION
**Pull request by mavduevskiy**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1879648

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 1]
```diff
@@ -410,7 +411,7 @@ open class DefaultToolbarMenu(
     val coreMenuItems by lazy {
         val menuItems =
             listOfNotNull(
-                if (shouldUseBottomToolbar) null else menuToolbar,
+                if (shouldUseBottomToolbar || shouldShowMenuToolbar) null else menuToolbar,
```
Comment (rocketsroger) [inline fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:414]: Should this be ```if (!shouldUseBottomToolbar && shouldShowMenuToolbar) menuToolbar else null,`?

**Comment by rocketsroger** [comment]
Looks like there are some test failures too.  Not sure if it's related to your change.

File: fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 1 above)
Comment (mavduevskiy) [reply fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: I was going for `if (shouldUseBottomToolbar || !shouldShowMenuToolbar)` but

**Comment by mavduevskiy** [comment]
I think tests failures were related to the wrong condition, I believe it was disabling the toolbar for some configurations. The value of having tests! Waiting for tests to run and re-requesting

**Comment by mavduevskiy** [comment]
Actually, a test was missing a `settings` dependency , updated!

**Comment by rocketsroger** [comment]
Looks like one more test needs to be fixed: `custom tab toolbar menu uses the proper custom tab session`

**Comment by mavduevskiy** [comment]
@rocketsroger updated

File: fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 1 above)
Comment (rocketsroger) [reply fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: Sure that works as well.  I just like the positive test cases for easier reading.

**Review by rocketsroger — APPROVED** [review]
🚢

**Pull request merged**